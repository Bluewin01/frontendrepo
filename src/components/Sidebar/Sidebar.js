import React from "react";
import {
  MdOutlineMailOutline,
  MdOutlineSms,
  MdNotificationImportant,
} from "react-icons/md";
import {
  Side,
  SidebarMenu,
  SidebarMenuIndicator,
  SidebarMenuItem,
} from "./SidebarElements";

const sidebarNavItems = [
  {
    display: "Email",
    icon: <MdOutlineMailOutline size={28} />,
  },
  {
    display: "Sms",
    icon: <MdOutlineSms size={28} />,
  },
  {
    display: "Push",
    icon: <MdNotificationImportant size={28} />,
  },
];

const Sidebar = (props) => {
  const { templateType } = props;

  return (
    <Side>
      <SidebarMenu>
        {sidebarNavItems.map((item, index) => (
          <SidebarMenuItem
            key={index}
            active={item.display === templateType ? true : false}
            onClick={() => {
              props.handleClick(item.display);
            }}
          >
            <div>{item.icon}</div>
            <div>{item.display}</div>
          </SidebarMenuItem>
        ))}
        <SidebarMenuIndicator type={templateType}></SidebarMenuIndicator>
      </SidebarMenu>
    </Side>
  );
};

export default Sidebar;
