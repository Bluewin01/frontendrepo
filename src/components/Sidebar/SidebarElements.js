import styled from "styled-components";

export const Side = styled.div`
  display: flex;
  flex-direction: column;
  flex-shrink: 0;
  width: 15%;
  padding: 5px 0px 0px 0px;
`;

export const SidebarMenu = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
`;

export const SidebarMenuItem = styled.button`
  width: 90%;
  padding: 10px 0px;
  margin: 5px 0px 0px 0px;
  font-weight: 500;
  transition: color 0.3s ease-in-out;
  border: none;
  background-color: transparent;
  color: ${(props) => (props.active ? "#fff" : "#555555")};
  ${(props) =>
    props.active
      ? ""
      : "&:hover { background-color: #ffe1e1; border-radius: 10px;}"}
`;

export const SidebarMenuIndicator = styled.div`
  position: absolute;
  height: 77px
  width: 90%;
  top: 5px;
  left: 50%;
  border-radius: 10px;
  background-color: #d31145;
  z-index: -1;
  transition: 0.3s ease-in-out;
  transform: ${(props) =>
    props.type === "Email"
      ? "translateX(-50%) translateY(0px)"
      : props.type === "Sms"
      ? "translateX(-50%) translateY(82px)"
      : "translateX(-50%) translateY(164px)"};
`;
