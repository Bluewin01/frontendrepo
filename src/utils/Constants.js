export const theme = {
  color: {
    w: "#FFFFFF",
    k: "#000000",
    p1: "#D31145",
    p2: "#596C80",
    p3: "#554344",
    p4: "#363E3F",
    p5: "#e6edeb",
    s1: "#E86487",
    s2: "#FBBF56",
    s3: "#97CB5D",
    s4: "#46D3BD",
    s5: "#22A8DA",
    s6: "#9962D2",
    t1: "#F6CCD7",
    t2: "#B8123E",
    t3: "#F9E1B8",
    t4: "#C36418",
    t5: "#C6EB9E",
    t6: "#65962E",
    b1: "#D4EDF1",
    b2: "#FAF9F3",
    b3: "#F0EEE1",
    b4: "#DBDAD4",
    b5: "#B3B1A8",
    b6: "#6F6E68",
    b8: "#302F2C",
    b9: "#292025",
    u2: "#FF5500",
    u3: "#405B70",
    u4: "#769FAB",
  },
};

export const BASE_API = `${process.env.REACT_APP_BASE_API_URL}`;
