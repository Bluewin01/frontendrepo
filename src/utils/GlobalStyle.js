import { createGlobalStyle } from 'styled-components';
import 'antd/dist/antd.css';
import '@aia-digital/ui-library/lib/static/styles.css';

import AIABoldEOT from '../assets/fonts/AIAEverest-Bold.eot';
import AIABoldWOFF from '../assets/fonts/AIAEverest-Bold.woff';
import AIABoldWOFF2 from '../assets/fonts/AIAEverest-Bold.woff2';
import AIABoldTTF from '../assets/fonts/AIAEverest-Bold.ttf';
import AIABoldSVG from '../assets/fonts/AIAEverest-Bold.svg';

import AIACondensedEOT from '../assets/fonts/AIAEverest-Condensed.eot';
import AIACondensedWOFF from '../assets/fonts/AIAEverest-Condensed.woff';
import AIACondensedWOFF2 from '../assets/fonts/AIAEverest-Condensed.woff2';
import AIACondensedTTF from '../assets/fonts/AIAEverest-Condensed.ttf';
import AIACondensedSVG from '../assets/fonts/AIAEverest-Condensed.svg';

import AIACondensedMediumEOT from '../assets/fonts/AIAEverest-CondensedMedium.eot';
import AIACondensedMediumWOFF from '../assets/fonts/AIAEverest-CondensedMedium.woff';
import AIACondensedMediumWOFF2 from '../assets/fonts/AIAEverest-CondensedMedium.woff2';
import AIACondensedMediumTTF from '../assets/fonts/AIAEverest-CondensedMedium.ttf';
import AIACondensedMediumSVG from '../assets/fonts/AIAEverest-CondensedMedium.svg';

import AIAExtraBoldEOT from '../assets/fonts/AIAEverest-ExtraBold.eot';
import AIAExtraBoldWOFF from '../assets/fonts/AIAEverest-ExtraBold.woff';
import AIAExtraBoldWOFF2 from '../assets/fonts/AIAEverest-ExtraBold.woff2';
import AIAExtraBoldTTF from '../assets/fonts/AIAEverest-ExtraBold.ttf';
import AIAExtraBoldSVG from '../assets/fonts/AIAEverest-ExtraBold.svg';

import AIAMediumEOT from '../assets/fonts/AIAEverest-Medium.eot';
import AIAMediumWOFF from '../assets/fonts/AIAEverest-Medium.woff';
import AIAMediumWOFF2 from '../assets/fonts/AIAEverest-Medium.woff2';
import AIAMediumTTF from '../assets/fonts/AIAEverest-Medium.ttf';
import AIAMediumSVG from '../assets/fonts/AIAEverest-Medium.svg';

import AIARegularEOT from '../assets/fonts/AIAEverest-Regular.eot';
import AIARegularWOFF from '../assets/fonts/AIAEverest-Regular.woff';
import AIARegularWOFF2 from '../assets/fonts/AIAEverest-Regular.woff2';
import AIARegularTTF from '../assets/fonts/AIAEverest-Regular.ttf';
import AIARegularSVG from '../assets/fonts/AIAEverest-Regular.svg';

const GlobalStyle = createGlobalStyle`
  @font-face {
    font-family: 'AIABold';
    src: url('${AIABoldEOT}');
    src: url('${AIABoldEOT}?#iefix') format('embedded-opentype'),
        url('${AIABoldWOFF}') format('woff'),
        url('${AIABoldWOFF2}') format('woff2'),
        url('${AIABoldTTF}') format('truetype'),
        url('${AIABoldSVG}#AIAEverestBold') format('svg');
    font-weight: normal;
    font-style: normal;
  }
  @font-face {
    font-family: 'AIACondensed';
    src: url('${AIACondensedEOT}');
    src: url('${AIACondensedEOT}?#iefix') format('embedded-opentype'),
        url('${AIACondensedWOFF}') format('woff'),
        url('${AIACondensedWOFF2}') format('woff2'),
        url('${AIACondensedTTF}') format('truetype'),
        url('${AIACondensedSVG}#AIAEverestCondensed') format('svg');
    font-weight: normal;
    font-style: normal;
  }
  @font-face {
    font-family: 'AIACondensedMedium';
    src: url('${AIACondensedMediumEOT}');
    src: url('${AIACondensedMediumEOT}?#iefix') format('embedded-opentype'),
        url('${AIACondensedMediumWOFF}') format('woff'),
        url('${AIACondensedMediumWOFF2}') format('woff2'),
        url('${AIACondensedMediumTTF}') format('truetype'),
        url('${AIACondensedMediumSVG}#AIAEverestCondensedMedium') format('svg');
    font-weight: 500;
    font-style: normal;
  }
  @font-face {
    font-family: 'AIAExtraBold';
    src: url('${AIAExtraBoldEOT}');
    src: url('${AIAExtraBoldEOT}?#iefix') format('embedded-opentype'),
        url('${AIAExtraBoldWOFF}') format('woff'),
        url('${AIAExtraBoldWOFF2}') format('woff2'),
        url('${AIAExtraBoldTTF}') format('truetype'),
        url('${AIAExtraBoldSVG}#AIAEverestExtraBold') format('svg');
    font-weight: normal;
    font-style: normal;
  }
  @font-face {
    font-family: 'AIAMedium';
    src: url('${AIAMediumEOT}');
    src: url('${AIAMediumEOT}?#iefix') format('embedded-opentype'),
        url('${AIAMediumWOFF}') format('woff'),
        url('${AIAMediumWOFF2}') format('woff2'),
        url('${AIAMediumTTF}') format('truetype'),
        url('${AIAMediumSVG}#AIAEverestMedium') format('svg');
    font-weight: 600;
    font-style: normal;
  }
  @font-face {
    font-family: 'AIARegular';
    src: url('${AIARegularEOT}');
    src: url('${AIARegularEOT}?#iefix') format('embedded-opentype'),
        url('${AIARegularWOFF}') format('woff'),
        url('${AIARegularWOFF2}') format('woff2'),
        url('${AIARegularTTF}') format('truetype'),
        url('${AIARegularSVG}#AIAEverestRegular') format('svg');
    font-weight: 400;
    font-style: normal;
  }

  body h1 {
    font-family: "AIACondensedMedium", "Calibri", sans-serif;
    font-size: 96px;
    line-height: 84px;
    font-weight: normal;
    margin-top: 0px;
    margin-bottom: 0px;
    color: ${({ theme }) => theme.color.p3}
  }

  body h2 {
    font-family: "AIACondensedMedium", "Calibri", sans-serif;
    font-size: 72px;
    line-height: 64px;
    font-weight: normal;
    margin-top: 0px;
    margin-bottom: 0px;
    color: ${({ theme }) => theme.color.p3}
  }

  body h3 {
    font-family: "AIACondensedMedium", "Calibri", sans-serif;
    font-size: 48px;
    line-height: 44px;
    font-weight: normal;
    margin-top: 0px;
    margin-bottom: 0px;
    color: ${({ theme }) => theme.color.p3}
  }

  body h4 {
    font-family: "AIACondensedMedium", "Calibri", sans-serif;
    font-size: 32px;
    line-height: 28px;
    font-weight: normal;
    margin-top: 0px;
    margin-bottom: 0px;
    color: ${({ theme }) => theme.color.p3}
  }

  body h5 {
    font-family: "AIACondensedMedium", "Calibri", sans-serif;
    font-size: 24px;
    line-height: 22px;
    font-weight: normal;
    margin-top: 0px;
    margin-bottom: 0px;
    color: ${({ theme }) => theme.color.p3}
  }

  body h6 {
    font-family: "AIACondensedMedium", "Calibri", sans-serif;
    font-size: 20px;
    line-height: 20px;
    font-weight: normal;
    margin-top: 0px;
    margin-bottom: 0px;
    color: ${({ theme }) => theme.color.p3}
  }

  body p, ul, ol {
    font-family: "AIARegular", Arial, "Helvetica Neue", Helvetica, sans-serif;
    font-size: 15px;
    line-height: 20px;
    margin-bottom: 0px;
    color: ${({ theme }) => theme.color.p3}
  }
`;

export default GlobalStyle;
