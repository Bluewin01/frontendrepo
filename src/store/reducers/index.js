import { combineReducers } from "redux";
import { configureStore } from "@reduxjs/toolkit";
import lang from "../Lang/LangReducer";
import loader from "../Loader/LoaderReducer";
import Auth from "../Auth/AuthReducer";
import Template from "../Template/TemplateReducer";

const rootReducer = combineReducers({
  lang,
  loader,
  Auth,
  Template,
});

export const setupStore = (preloadedState) => {
  return configureStore({
    reducer: rootReducer,
    preloadedState,
  });
};

export default rootReducer;
