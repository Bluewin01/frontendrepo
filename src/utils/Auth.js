// Service to check authentication for user and to signOut
import * as AuthActions from "../store/Auth/AuthActions";
import store from "../store";

const Auth = {
  signOut() {
    store.dispatch(AuthActions.signOut);
  },
  isAuth(jwt) {
    if (!jwt) {
      return false;
    } else {
      return true;
    }
  },
};
export default Auth;
