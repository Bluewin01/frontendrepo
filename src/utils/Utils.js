// test-utils.jsx
import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import { setupStore } from "../store/reducers";

export function render(
  ui,
  { preloadedState, store = setupStore(preloadedState), ...renderOptions } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...rtlRender(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from "@testing-library/react";
