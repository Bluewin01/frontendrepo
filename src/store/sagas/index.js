import { AuthSaga } from "../Auth/AuthSaga";
import { TemplateSaga } from "../Template/TemplateSaga";
import { fork, all } from "redux-saga/effects";

export function* watchSagas() {
  yield all([AuthSaga(), TemplateSaga()]);
}
