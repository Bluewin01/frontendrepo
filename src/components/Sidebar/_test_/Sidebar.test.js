import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { cleanup } from "../../../utils/Utils";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import store from "../../../store";
import { Provider } from "react-redux";
import Sidebar from "../Sidebar";

describe("Component: Sidebar", () => {
  beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // Deprecated
        removeListener: jest.fn(), // Deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });
  afterEach(cleanup);

  it("should be render correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Sidebar />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
