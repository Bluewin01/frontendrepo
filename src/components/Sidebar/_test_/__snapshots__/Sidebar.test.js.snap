// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Sidebar should be render correctly 1`] = `
<div
  className="sc-bdVaJa efNFrH"
>
  <div
    className="sc-bwzfXH kaXCHS"
  >
    <button
      className="sc-htpNat gJvYaN"
      onClick={[Function]}
    >
      <div>
        <svg
          fill="currentColor"
          height={28}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width={28}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V8l8 5 8-5v10zm-8-7L4 6h16l-8 5z"
          />
        </svg>
      </div>
      <div>
        Email
      </div>
    </button>
    <button
      className="sc-htpNat gJvYaN"
      onClick={[Function]}
    >
      <div>
        <svg
          fill="currentColor"
          height={28}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width={28}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M20 2H4c-1.1 0-2 .9-2 2v18l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 14H5.17L4 17.17V4h16v12zM7 9h2v2H7zm8 0h2v2h-2zm-4 0h2v2h-2z"
          />
        </svg>
      </div>
      <div>
        Sms
      </div>
    </button>
    <button
      className="sc-htpNat gJvYaN"
      onClick={[Function]}
    >
      <div>
        <svg
          fill="currentColor"
          height={28}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width={28}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M18 16v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2zm-5 0h-2v-2h2v2zm0-4h-2V8h2v4zm-1 10c1.1 0 2-.9 2-2h-4a2 2 0 002 2z"
          />
        </svg>
      </div>
      <div>
        Push
      </div>
    </button>
    <div
      className="sc-bxivhb hpkBHJ"
    />
  </div>
</div>
`;
